yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -103,11 +104,15 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
           userClass = value
           args = tail
 
+        case ("--primaryResource") :: value :: tail =>
```

Comment:
What's the need for this new argument?  The argument that's not an option is implicitly understood to be the primary resource.  Is there a reason this won't work for python?